_user/rotrectangle/main.c

```diff
@@ -0,0 +1,273 @@
+ * Phoenix-RTOS
+ *
+ * Rotating Rectangle
+ *
+ * Example of user application
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+/*					a
+*	(x0,y0)	.................
+*			|				|
+*			|		.		|
+*		b	|	(xc,yc)		|
+*			|...............|
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <graph.h>
+#include <time.h>
+#include <unistd.h>
+#include <math.h>
+#include <termios.h>
+
+#include <sys/threads.h>
+
+typedef struct {
+	unsigned int xc;
+	unsigned int yc;
+	unsigned int a;
+	unsigned int b;
+	unsigned int color;
+} rectangle_t;
+
+typedef struct {
+	rectangle_t rec;
+	rectangle_t toRemove;
+	unsigned int rotSpeed;
+	unsigned int maxxc;
+	unsigned int maxyc;
+	unsigned int minxc;
+	unsigned int minyc;
+	char stop;
+} rotrectangle_t;
+
+/* Enables/disables canon mode and echo */
+static int rotrectangle_switchmode(int canon)
+{
+	struct termios state;
+
+	tcgetattr(STDIN_FILENO, &state);
+	if (canon) {
+		state.c_lflag |= ICANON;
+		state.c_lflag |= ECHO;
+	} else {
+		state.c_lflag &= ~ICANON;
+		state.c_lflag &= ~ECHO;
+		state.c_cc[VMIN] = 1;
+	}
+	tcsetattr(STDIN_FILENO, TCSANOW, &state);
+	
+	return 0;
+}
+
+
+static void rotrectangle_getkey(void *arg)
+{
+	char ch;
+	rotrectangle_t *self = (rotrectangle_t *)arg;
+	while ((self->stop == 0) && (ch = getchar())) {
+		if (ch == 27) {					/*ESC code from arrow keys*/
+			getchar();
+			ch = getchar();
+		}
+		switch (ch) {
+			case 67:
+				if (self->rec.xc < (self->maxxc-2))
+					self->rec.xc+=3;
+				break;
+			case 68:
+				if (self->rec.xc > (self->minxc+2))
+					self->rec.xc-=3;
+				break;
+			case 65:
+				if (self->rec.yc > (self->minyc+2))
+					self->rec.yc-=3;
+				break;
+			case 66:
+				if (self->rec.yc < (self->maxyc-2))
+					self->rec.yc+=3;
+				break;
+			case 43:
+				if (self->rotSpeed < 5)
+					self->rotSpeed++;
+				break;
+			case 45:
+				if (self->rotSpeed > 1)
+					self->rotSpeed--;
+				break;
+			case 'r':
+				self->rec.color = 4;
+				break;
+			case 'g':
+				self->rec.color = 2;
+				break;
+			case 'b':
+				self->rec.color = 1;
+				break;
+			case 'q':
+				self->stop = 1;
+				break;
+			default:
+				break;
+		}
+	}
+}
+
+static int rotrectangle_print(rectangle_t *self, graph_t *graphp, unsigned int angle)
+{
+	double dx, dy, angleRad, xcDouble, ycDouble, aDouble, bDouble;
+	unsigned int x0, y0, x, y;		/* (x,y) is start point for next rectangle lines */
+
+	xcDouble = (double)self->xc;
+	ycDouble = (double)self->yc;
+	aDouble = (double)self->a;
+	bDouble = (double)self->b;
+
+	angleRad = ((double)(angle)*3.141592)/180.0;
+	x0 = (unsigned int)(xcDouble - ( (sin(M_PI/2-angleRad-atan(bDouble/aDouble)))*sqrt(pow(aDouble, 2.0)+pow(bDouble, 2.0)) )/2.0);
+	y0 = (unsigned int)(ycDouble - ( (cos(M_PI/2-angleRad-atan(bDouble/aDouble)))*sqrt(pow(aDouble, 2.0)+pow(bDouble, 2.0)) )/2.0);
+
+	x = x0;
+	y = y0;
+	dx = (self->a)*cos(angleRad); 
+	dy = (self->a)*sin(angleRad);
```

Comment:
There should be space between `*`, `+`, etc. operands, e.g.:
`dx = (self->a) * cos(angleRad);`

Applies to other occurrences